File: app/src/main/java/net/sytes/kai_soft/letsbuyka/Lists/DetailFragmentList.java
package net.sytes.kai_soft.letsbuyka.Lists;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import net.sytes.kai_soft.letsbuyka.CRUDdb;
import net.sytes.kai_soft.letsbuyka.IMenuContract;
import net.sytes.kai_soft.letsbuyka.R;
import net.sytes.kai_soft.letsbuyka.IListActivityContract;

/**
 * Created by Лунтя on 06.06.2018.
 */

public class DetailFragmentList extends Fragment implements IMenuContract {

    //Button insertBtn, backToListBtn, deleteBtn;
    EditText etName;
    List list;
    //DataBase dbProduct;
    IListActivityContract iListActivityContract;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_lists_detail, container, false);

        /*insertBtn = rootView.findViewById(R.id.insertListBtn);
        backToListBtn = rootView.findViewById(R.id.backToListListBtn);
        deleteBtn = rootView.findViewById(R.id.deleteListBtn);*/

        etName = rootView.findViewById(R.id.etNameList);


        //dbProduct = Application.getDB(); //new DataBase(this.getContext());

        return rootView;
    }

    @Override
    public void onResume() {
        super.onResume();

        makeEditable(true);

        emptyEditText();

        /*backToListBtn.setOnClickListener(this);
        insertBtn.setOnClickListener(this);
        deleteBtn.setVisibility(View.GONE);*/

        Bundle bundle = getArguments();

        if (bundle.getBoolean("editable")) {
            //insertBtn.setText(R.string.save);

        } else {
            list = (List) bundle.getSerializable("list");

            etName.setText(list.getItemName());


            makeEditable(true);

            /*insertBtn.setText(R.string.edit);

            deleteBtn.setOnClickListener(this);
            deleteBtn.setVisibility(View.VISIBLE);*/
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof IListActivityContract) {
            iListActivityContract = (IListActivityContract) context;
        }
    }


    @Override
    public void onDetach() {
        iListActivityContract = null;
        super.onDetach();
    }
    private boolean isEditable() {
        return etName.isEnabled();
    }

    private void makeEditable(boolean state) {
        etName.setEnabled(state);
    }

    private void emptyEditText() {
        etName.setText("");
    }

    private boolean isNewElement() {
        Bundle bundle = getArguments();
        return bundle.getBoolean("editable") == true;
    }

    private List toUpdate() {
        Bundle bundle = getArguments();
        List fromBundle = (List) bundle.getSerializable("list");
        List toUpdate = new List(fromBundle.getId(),
                etName.getText().toString());
        return toUpdate;
    }

    @Override
    public void savePressed() {
        if (isNewElement() == false) {
            List updateble = toUpdate();
            CRUDdb.Companion.updateTableLists(updateble);
            iListActivityContract.onDetailFragmentButtonClick();
        } else {
            CRUDdb.Companion.insertToTableLists(etName.getText().toString());
            iListActivityContract.onDetailFragmentButtonClick();
        }
    }
    @Override
    public void deletePressed(){
        Bundle bundle = getArguments();
        CRUDdb.Companion.deleteItemLists((List) bundle.getSerializable("list"));
        iListActivityContract.onDetailFragmentButtonClick();
    }
}

File: app/src/main/java/net/sytes/kai_soft/letsbuyka/Lists/DetailFragmentList.kt
package net.sytes.kai_soft.letsbuyka.Lists

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

import net.sytes.kai_soft.letsbuyka.CRUDdb
import net.sytes.kai_soft.letsbuyka.IMenuContract
import net.sytes.kai_soft.letsbuyka.R
import net.sytes.kai_soft.letsbuyka.IActivityContract

/*  Detail fragment for selected list   */
class DetailFragmentList: Fragment(), IMenuContract {
    companion object {
        private const val myTag = "letsbuy_LDetailFragment"
    }

    private lateinit var etName: EditText
    private lateinit var list: List
    private lateinit var iActivityContract: IActivityContract

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.i(myTag, "onCreateView()")
        val rootView = inflater!!.inflate(R.layout.fragment_lists_detail, container, false)
        etName = rootView.findViewById(R.id.etNameList)

        return rootView
    }

    override fun onResume() {
        Log.i(myTag, "onResume()")
        super.onResume()
        emptyEditText()

        val bundle = arguments
        if (!bundle.getBoolean("editable")){
            list = bundle.getSerializable("list") as List
            etName.setText(list.itemName)
        }
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is IActivityContract){
            iActivityContract = context
        }
    }

    private fun emptyEditText(){
        etName.text.clear()
    }

    private fun isNewElement(): Boolean{
        val bundle = arguments
        return bundle.getBoolean("editable")
    }

    private fun toUpdate(): List {
        val bundle = arguments
        val fromBundle = bundle.getSerializable("list") as List
        return List(fromBundle.id, etName.text.toString())
    }

    override fun savePressed() {
        if (!isNewElement()){
            val updatable = toUpdate()
            CRUDdb.updateTableLists(updatable)
            iActivityContract.onDetailFragmentButtonClick()
        } else {
            CRUDdb.insertToTableLists(etName.text.toString())
            iActivityContract.onDetailFragmentButtonClick()
        }
    }

    override fun deletePressed() {
        val bundle = arguments
        CRUDdb.deleteItemLists(bundle.getSerializable("list") as List)
        iActivityContract.onDetailFragmentButtonClick()
    }

    override fun hideKeyboard() {
        val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(etName.windowToken, 0)
    }
}